### DISCUSSION TITLE
Rolling back changes with git does not restore the repo to its orignal state

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#11153] [Q&A] [answered]
Clone [this repo](https://github.com/sam-wheat/UnoTest.git).  Repo consists of a solution with an unmodified Uno app and unmodified Uno class library.
Make a copy in a different temp folder (optional).
Build and run the app.
Update all the nuget packages for all the projects.  The solution will not build and some the projects will not load.
Use git to undo all the changes in the repo.
At this point the repository should be back in it's original state - however the project will no longer build (restarting VS and deleting the .vs folder does not work).  See error messages below.

Delete all files and folders including .git.
Clone the repo again in the exact same folder. 
The app builds and runs (suggesting no change or corruption in an external file).

I used a directory compare tool to compare the post-undo repo to a backup.  The compare tool finds no differences.  File differ finds the file encoding has been changed to UTF-8.  However making an arbitrary change to a .csproj file and undoing with git seems to work.

I tried this exercise using just a Uno App - no class lib - I can not reproduce the problem.

Git reports the files below exist outside the repo - are they changed when nuget packages are updated?  The fact that I can delete the repo and clone again and it works leads me to believe that external files are not being changed or corrupted.

> C:\Users\sam\.nuget\packages\uno.wasm.bootstrap\7.0.11\contentFiles\any\netstandard2.0\obj\uno-bootstrap.d.ts,

> C:\Users\sam\.nuget\packages\uno.wasm.bootstrap\7.0.11\contentFiles\any\netstandard2.0\obj\uno-bootstrap.js

Error messages:

> NU1101	Unable to find package Microsoft.NETCore.App.Runtime.Mono.win-arm64. No packages exist with this id in source(s): C:\Program Files\dotnet\library-packs

> APPX3217	SDK folder containing 'UAP.props' for 'UAP 16.1' cannot be located.

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the report. This is a known issue in the `Microsoft.WindowsAppSDK` package (https://github.com/microsoft/WindowsAppSDK/issues/2338), where there's a invalid condition that happens when referenced unconditionally in a class library:

```
A numeric comparison was attempted on "$(TargetPlatformVersion)" that evaluates to ""
instead of a number, in condition "'$(TargetPlatformVersion)' < '10.0.18362.0'"
```

 This state can happen when using the NuGet package manager to update a csproj, which causes incorrect `<PackageReference />` to be unconditionally included.

To fix the problem, close all your visual studio instances (so that no `msbuild.exe` stays running), delete the `bin` and `obj` folders and without opening visual studio, and make sure to edit the csproj to remove any unconditional reference to `Microsoft.WindowsAppSDK`.

The csproj should look like this file, and particularly this block:
https://github.com/unoplatform/uno/blob/a911ec8288090db0cf3fd0c6a8815c612e531ab7/src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Views/UnoQuickStart.csproj#L35-L39

Where the `Microsoft.WindowsAppSDK` reference should only be present once, and only inside the block `<ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.18362'">`.

**Reply by jeromelaban** [reply]
Here's the visual studio feedback issue, if you want to upvote it: https://developercommunity.visualstudio.com/t/NuGet-Package-manager-does-not-honor-con/10269291

**Reply by sam-wheat** [reply]
Thank you @jeromelaban I am still interested to know why the modified .csproj files are not rolled back or do not show as modified in git.  Is VS using cached copies on subsequent builds?

**Reply by jeromelaban** [reply]
Yes VS is sometimes incorrectly caching files, which is why closing visual studio is an important step as well as deleting bin/obj files.
